ext.go

```diff
@@ -50,6 +50,16 @@ func registerExt(id int8, typ reflect.Type, enc encoderFunc, dec decoderFunc) {
 	}
 }
 
+func (e *Encoder) EncodeExtensionHeader(typeId byte, length int) error {
```

Comment:
There is already `RegisterExt` so this should be called `EncodeExtHeader(typeId int8, length int) error` for consistency (note byte -> int8 change).